core/delete_area.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The abstract class for a component that can delete a block that
+ *   is dropped on top of it.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+'use strict';
+
+goog.provide('Blockly.DeleteArea');
+
+goog.require('Blockly.IDragTarget');
+
+goog.requireType('Blockly.utils.Rect');
+
+
+/**
+ * Abstract class for a component that can delete a block that is dropped on top
+ * of it.
+ * @implements {Blockly.IDragTarget}
+ * @constructor
+ */
+Blockly.DeleteArea = function() {};
```

Comment:
I haven't been registering it here because the id needs to be unique and we don't yet have support for adding capabilities. There is a PR out for that now: https://github.com/google/blockly/pull/4875

Going down this route, the ids will need to either:
- be randomly generated, so they wont be as descriptive.
- would need to be added as a property and set by the extending class (to allow for something like the trashcan to have the id "trashcan")

In summary, not enough support to do this yet. I'm open to adding this, but we'd have to consider the drawbacks of the id.